heroku/deploy-a-django-app.md

```diff
@@ -0,0 +1,272 @@
+# Deploy a Django app to Heroku
+
+This guide provides instructions on deploying a Django app to Heroku, DataMade's
+preferred platform for hosting dynamic applications.
+
+## Contents
+
+- [Set up application code for Heroku](#set-up-application-code-for-heroku)
+    - [Containerize your app](#containerize-your-app)
+    - [Serve static files with WhiteNoise](#serve-static-files-with-whitenoise)
+    - [Read settings and secret variables from the environment](#read-settings-and-secret-variables-from-the-environment)
+    - [Configure Django loggine](#configure-django-logging)
+- [Provision Heroku resources](#provision-heroku-resources)
+    - [Ensure that the Heroku CLI is installed](#ensure-that-the-heroku-cli-is-installed)
+    - [Create Heroku config files](#create-heroku-config-files)
+        - [`heroku.yml`](#heroku.yml)
+        - [`release.sh`](#release.sh)
+        - [`app.json`](#app.json)
+    - [Create apps and pipelines for your project](#create-apps-and-pipelines-for-your-project)
+- [Troubleshooting](#troubleshooting)
+
+## Set up application code for Heroku
+
+In order to deploy a Django application to Heroku, a few specific configurations
+need to be enabled in your application code. We provide details on these
+configurations below.
+
+### Containerize your app
+
+We use Heroku as a platform for deploying containerized apps, which means that
+your app must be containerized in order to use Heroku properly. If your app
+is not yet containerized, [follow our instructions for containerizing Django
+apps](/docker/local-development.md) before moving on.
+
+### Serve static files with WhiteNoise
+
+Apps deployed on Heroku don't typically use Nginx to serve content, so they need some
+other way of serving static files. Since our apps tend to have relatively low traffic,
+we prefer configuring [WhiteNoise](http://whitenoise.evans.io/)
+to allow Django to serve static files in production.
+
+Follow the [setup instructions for WhiteNoise in
+Django](http://whitenoise.evans.io/en/stable/django.html) to ensure that your
+Django apps can serve static files.
+
+In order to be able to serve static files when `DEBUG` is `False`, you'll also want
+to make sure that `RUN python manage.py collectstatic` is included as a step in your
+Dockerfile. This will ensure that the static files are baked into the container.
+Since we typically mount application code into the container during local development,
+you'll also need to make sure that your static files are stored outside of the root
+project folder so that the mounted files don't overwrite them. One easy way to do this
+is to set `STATIC_ROOT = '/static'` in your `settings.py` file.
+
+### Read settings and secret variables from the environment
```

Comment:
Your [elections archive settings file](https://github.com/datamade/mn-election-archive/blob/master/elections/settings.py) is a master class in this exercise! I cribbed really heavily from it [for Metro](https://github.com/datamade/la-metro-councilmatic/blob/2.5_deploy/councilmatic/settings.py). A link to one or both examples would be a great addition.

Reply:
Thank you! I added links to my settings and Compose files here, and then added links to Metro later on in the Websolr section. Is the `2.5_deploy` branch stable enough that we can link to it permanently here?